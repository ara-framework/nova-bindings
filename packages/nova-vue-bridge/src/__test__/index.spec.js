import Vue from 'vue';

import Nova from '..';

jest.mock('uuid/v4', () => () => 'uuid');

describe('nova-vue-bridge', () => {
  beforeEach(() => {
    document.body.innerHTML = '<div id="app"></div>';
  });

  test('should render placeholder correctly', () => {
    const Component = Vue.extend(Nova);

    const vm = new Component({
      propsData: {
        name: 'Example',
        data: {
          title: 'Nova Bridge',
        },
      },
    });

    vm.$mount(document.getElementById('app'));

    expect(document.body.innerHTML).toMatchSnapshot();
  });

  test('should render placeholder when is updated', async () => {
    const Component = Vue.extend(Nova);

    const vm = new Component({
      propsData: {
        name: 'Example',
        data: {
          title: 'Nova Bridge',
        },
      },
    });

    vm.$mount(document.getElementById('app'));

    await new Promise((resolve) => {
      vm.$nextTick(() => {
        resolve();
      });
      vm.data = { title: 'Nova Bridge Updated' };
    });
    expect(document.body.innerHTML).toMatchSnapshot();
  });

  test('should emit event when placeholder is mounted', async () => {
    const Component = Vue.extend(Nova);

    const vm = new Component({
      propsData: {
        name: 'Example',
        data: {
          title: 'Nova Bridge',
        },
      },
    });

    const event = await new Promise((resolve) => {
      document.addEventListener('NovaMount', (evt) => {
        resolve(evt);
      });

      vm.$mount(document.getElementById('app'));
    });

    expect(event.detail).toMatchSnapshot();
  });

  test('should emit event when placeholder is updated', async () => {
    const Component = Vue.extend(Nova);

    const vm = new Component({
      propsData: {
        name: 'Example',
        data: {
          title: 'Nova Bridge',
        },
      },
    });

    const event = await new Promise((resolve) => {
      vm.$mount(document.getElementById('app'));

      document.addEventListener('NovaMount', (evt) => {
        resolve(evt);
      });

      vm.data = { title: 'Nova Bridge Updated' };
    });

    expect(event.detail).toMatchSnapshot();
  });
});
