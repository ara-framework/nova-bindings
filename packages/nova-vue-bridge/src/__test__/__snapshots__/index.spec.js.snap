// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`nova-vue-bridge should emit event when placeholder is mounted 1`] = `
Object {
  "id": "uuid",
  "name": "Example",
}
`;

exports[`nova-vue-bridge should emit event when placeholder is updated 1`] = `
Object {
  "id": "uuid",
  "name": "Example",
}
`;

exports[`nova-vue-bridge should render placeholder correctly 1`] = `"<div><div data-hypernova-key=\\"Example\\" data-hypernova-id=\\"uuid\\"></div><script type=\\"application/json\\" data-hypernova-key=\\"Example\\" data-hypernova-id=\\"uuid\\"><!--{\\"title\\":\\"Nova Bridge\\"}--></script></div>"`;

exports[`nova-vue-bridge should render placeholder when is updated 1`] = `"<div><div data-hypernova-key=\\"Example\\" data-hypernova-id=\\"uuid\\"></div><script type=\\"application/json\\" data-hypernova-key=\\"Example\\" data-hypernova-id=\\"uuid\\"><!--{\\"title\\":\\"Nova Bridge Updated\\"}--></script></div>"`;
